ingest/scripts/get_loculus_submissions.py

```diff
@@ -0,0 +1,182 @@
+import logging
+import os
+from dataclasses import dataclass
+from datetime import datetime
+from enum import Enum
+
+import click
+import yaml
+from psycopg2.extras import RealDictCursor
+from psycopg2.pool import SimpleConnectionPool
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(
+    encoding="utf-8",
+    level=logging.DEBUG,
+    format="%(asctime)s %(levelname)8s (%(filename)20s:%(lineno)4d) - %(message)s ",
+    datefmt="%H:%M:%S",
+)
+
+
+@dataclass
+class Config:
+    segmented: bool
+    db_password: str
+    db_username: str
+    db_host: str
+
+
+def db_init(
+    db_password_default: str, db_username_default: str, db_host_default: str
+) -> SimpleConnectionPool:
+    db_password = os.getenv("DB_PASSWORD")
+    if not db_password:
+        db_password = db_password_default
+
+    db_username = os.getenv("DB_USERNAME")
+    if not db_username:
+        db_username = db_username_default
+
+    db_host = os.getenv("DB_HOST")
+    if not db_host:
+        db_host = db_host_default
+
+    return SimpleConnectionPool(
+        minconn=1,
+        maxconn=1,  # Only allow one connection per organism
+        dbname="loculus",
+        user=db_username,
+        host=db_host,
+        password=db_password,
+        options="-c search_path=ena-submission",
+    )
+
+
+class Status(Enum):
+    READY = 0
+    SUBMITTING = 1
+    SUBMITTED = 2
+    HAS_ERRORS = 3
+    WAITING = 4  # Only for assembly creation
+
+    def __str__(self):
+        return self.name
+
+
+@dataclass
+class SampleTableEntry:
+    accession: str
+    version: int
+    errors: str | None = None
+    warnings: str | None = None
+    status: Status = Status.READY
+    started_at: datetime | None = None
+    finished_at: datetime | None = None
+    result: str | None = None
+
+
+@dataclass
+class AssemblyTableEntry:
+    accession: str
+    version: int
+    errors: str | None = None
+    warnings: str | None = None
+    status: Status = Status.READY
+    started_at: datetime | None = None
+    finished_at: datetime | None = None
+    result: str | None = None
+
+
+def get_bio_sample_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
+    con = db_conn_pool.getconn()
+    try:
+        with con, con.cursor(cursor_factory=RealDictCursor) as cur:
+            # Result is a jsonb column
+            query = "SELECT accession, result FROM sample_table WHERE STATUS = 'SUBMITTED'"
+
+            cur.execute(query)
+
+            results = cur.fetchall()
+    finally:
+        db_conn_pool.putconn(con)
+
+    return {result["accession"]: result["result"]["biosample_accession"] for result in results}
```

Comment:
```suggestion
def get_bio_sample_accessions(db_conn_pool: SimpleConnectionPool) -> dict[str, str]:
    with db_conn_pool.getconn() as con:
        with con.cursor(cursor_factory=RealDictCursor) as cur:
            query = "SELECT accession, result FROM sample_table WHERE STATUS = 'SUBMITTED'"
            cur.execute(query)
            results = cur.fetchall()
    return {result["accession"]: result["result"]["biosample_accession"] for result in results}
```

Reply:
Ah so from my research we need `db_conn_pool.putconn(con)` - if the connection breaks off or somehow the execution fails and we don't do this connections will not be added back to the pool and we will end up with not enough ephermeral ports for connecting to the db.

Normally this is not required with pyscopy (as soon as you connect with the `with ... as con:` statement it should handle closing the connection if this fails) but as I now use a connection pool this is required.

Reply:
ah but maybe we just update and in psycopg3 they do it for you: https://www.psycopg.org/psycopg3/docs/advanced/pool.html#basic-connection-pool-usage

Reply:
Thanks - sorry for bad advice! It seemed to me like this might be possible and then Claude agreed with me and I wrongly trusted it :) (But further research, plus ChatGPT, agrees with you!)

Reply:
Added an issue to update: https://github.com/loculus-project/loculus/issues/2879